### PR TITLE
Support alternate identifiers schemes (from uuid)

### PR DESCRIPTION
**Pull request by jessemyers**
Includes a naive example using a content-based addressing (hash) scheme.

### CODE REVIEW

File: microcosm_flask/namespaces.py [hunk 1]
```diff
@@ -33,15 +33,14 @@ class Namespace(object):
...
-        enable_basic_auth=False
-    ):
+    def __init__(self,
+                 subject,
+                 object_=None,
+                 path=None,
+                 controller=None,
+                 version=None,
+                 enable_basic_auth=False,
+                 identifier_type="uuid"):
```
Comment (jessemyers) [inline microcosm_flask/namespaces.py]: The naming conventions that have an identifier type now *require* it as an argument. I couldn't find any examples where this was a breaking change. The default identifier type remains `uuid`.

File: microcosm_flask/tests/test_custom_identifier.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+from microcosm_flask.operations import Operation
+from microcosm_flask.tests.conventions.fixtures import (
+    Person,
+    PersonSchema,
+    PERSON_1,
+    PERSON_2,
+)
+
+
+class ContentBasedAddressConverter(BaseConverter):
```
Comment (jessemyers) [inline microcosm_flask/tests/test_custom_identifier.py]: This part supports `cba:<id>` addresses.

File: microcosm_flask/tests/test_custom_identifier.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }, sort_keys=True)
+        return sha256(content).hexdigest()
+
+
+@binding("cba")
+def configure_hash_converter(graph):
+    graph.flask.url_map.converters["cba"] = ContentBasedAddressConverter
+
+
+PEOPLE = [
```
Comment (jessemyers) [inline microcosm_flask/tests/test_custom_identifier.py]: This part creates a simple person query by (the new) id.

File: microcosm_flask/tests/test_custom_identifier.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        func=retrieve_person,
+        response_schema=PersonSchema(),
+    ),
+}
+
+
+class TestIdentifierType(object):
+
+    def setup(self):
+        loader = load_from_dict(
```
Comment (jessemyers) [inline microcosm_flask/tests/test_custom_identifier.py]: Register the new converter. The test on L84 demonstrates that this is additive (on top of `uuid`)

**Comment by adamhadani** [comment]
LGTM

**Pull request closed without merging**